framework/src/main/java/org/checkerframework/common/accumulation/AccumulationTransfer.java

```diff
@@ -0,0 +1,140 @@
+package org.checkerframework.common.accumulation;
+
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.Tree.Kind;
+import java.util.Arrays;
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/**
+ * The default transfer function for an accumulation checker.
+ *
+ * <p>Subclasses should call the {@link #accumulate(Node, TransferResult, String...)} accumulate}
+ * method to accumulate a string at a particular program point.
+ */
+public class AccumulationTransfer extends CFTransfer {
+
+    /** The type factory. */
+    protected final AccumulationAnnotatedTypeFactory typeFactory;
+
+    /**
+     * Build a new AccumulationTransfer for the given analysis.
+     *
+     * @param analysis the analysis
+     */
+    public AccumulationTransfer(CFAnalysis analysis) {
+        super(analysis);
+        typeFactory = (AccumulationAnnotatedTypeFactory) analysis.getTypeFactory();
+    }
+
+    /**
+     * Updates the estimate of how many things {@code node} has accumulated.
+     *
+     * <p>If the node is an invocation of a method that returns its receiver, then its receiver's
+     * type will also be updated. This chain will continue as long as each receiver is itself a
+     * receiver-returning method invocation.
+     *
+     * <p>For example, suppose the argument is the expression {@code a.b().c()}, the new value is
+     * "foo", and b and c return their receiver (and are deterministic). Then all of the expressions
```

Comment:
Why is it important to mention that the method is deterministic?
I think every method that returns its receiver is deterministic, so this would be a vacuous test and it's confusing why it is mentioned as an extra condition.

Reply:
I think Manu made this point at some point, which is why we added it to the documentation. I agree it is distracting. @msridhar do you think removing any mention of determinism is okay here - it is a general fact about how dataflow works rather than something specific to how accumulation analysis works.

Reply:
Yes we can remove the mention of determinism